go/common/gethutil/gethutil.go

```diff
@@ -20,14 +19,7 @@ var EmptyHash = gethcommon.Hash{}
 // LCA - returns the latest common ancestor of the 2 blocks or an error if no common ancestor is found
 // it also returns the blocks that became canonical, and the once that are now the fork
 func LCA(ctx context.Context, newCanonical *types.Block, oldCanonical *types.Block, resolver storage.BlockResolver) (*common.ChainFork, error) {
-	b, cp, ncp, err := internalLCA(ctx, newCanonical, oldCanonical, resolver, []common.L1BlockHash{}, []common.L1BlockHash{oldCanonical.Hash()})
-	// remove the common ancestor
-	if len(cp) > 0 {
-		cp = cp[0 : len(cp)-1]
-	}
-	if len(ncp) > 0 {
-		ncp = ncp[0 : len(ncp)-1]
-	}
+	b, cp, ncp, err := internalLCA(ctx, newCanonical, oldCanonical, resolver, []common.L1BlockHash{}, []common.L1BlockHash{})
```

Comment:
this was not returningn the new hew head block on the canonical path.
Which was having some consequences when setting the "is_canonical" flag, when a fork was cancelled out